.github/workflows/shippable_builds.yml

```diff
@@ -792,6 +828,68 @@ jobs:
           releaseFiles: |
             uploads/${{ steps.rename.outputs.PKG_FILE }}
 
+      - name: Determine FTP Destination
+        id: ftp_destination
+        if: ${{ !inputs.skipFtp && contains(matrix.releaseTarget, 'ftp') && matrix.packageFormat == 'apk' }}
+        shell: bash
+        env:
+          RELEASE_TYPE: ${{ env.RELEASE_TYPE }}
+          VERSION_NAME: ${{ steps.pkginfo.outputs.VERSION_NAME }}
+          UPLOAD_FTP_STAGE: ${{ inputs.uploadFtpStage }}
+        run: |
+          GCP_ENV=prod
+          GCP_PROJECT=moz-fx-productdelivery-pr-38b5
+          FTP_URL_PREFIX="https://ftp.mozilla.org/pub/thunderbird-mobile/android/"
+          if [[ "$UPLOAD_FTP_STAGE" = "true" ]]; then
+            GCP_ENV=stage
+            GCP_PROJECT=moz-fx-productdelivery-no-7d6a
+            FTP_URL_PREFIX="https://ftp.stage.mozaws.net/pub/thunderbird-mobile/android/"
+          fi
+          SERVICE_ACCOUNT="thunderbird-team-${GCP_ENV}@${GCP_PROJECT}.iam.gserviceaccount.com"
+          WORKLOAD_IDENTITY_PROVIDER="projects/324168772199/locations/global/workloadIdentityPools/github-actions/providers/github-actions"
+
+          FTP_SUBDIR="releases"
+          if [[ "${RELEASE_TYPE}" = "daily" ]]; then
+            FTP_SUBDIR="nightly"
+          fi
+
+          echo "SERVICE_ACCOUNT=${SERVICE_ACCOUNT}" >> $GITHUB_OUTPUT
+          echo "WORKLOAD_IDENTITY_PROVIDER=${WORKLOAD_IDENTITY_PROVIDER}" >> $GITHUB_OUTPUT
+          echo "FTP_DESTINATION=${GCP_PROJECT}-productdelivery/pub/thunderbird-mobile/android/${FTP_SUBDIR}/${VERSION_NAME}" >> $GITHUB_OUTPUT
+          echo "FTP_URL_PREFIX=${FTP_URL_PREFIX}" >> $GITHUB_OUTPUT
+
+      - name: Auth to GCS for FTP
+        if: ${{ !inputs.skipFtp && contains(matrix.releaseTarget, 'ftp') && matrix.packageFormat == 'apk' }}
+        uses: google-github-actions/auth@v2
+        with:
+          token_format: 'access_token'
+          service_account: ${{ steps.ftp_destination.outputs.SERVICE_ACCOUNT }}
+          workload_identity_provider: ${{ steps.ftp_destination.outputs.WORKLOAD_IDENTITY_PROVIDER }}
+
+      - name: GCS Upload of APK Package to FTP
+        id: publish_ftp_pkg
+        if: ${{ !inputs.skipFtp && contains(matrix.releaseTarget, 'ftp') && matrix.packageFormat == 'apk' }}
+        uses: google-github-actions/upload-cloud-storage@v2
+        with:
+          path: uploads/${{ steps.rename.outputs.PKG_FILE }}
+          destination: ${{ steps.ftp_destination.outputs.FTP_DESTINATION }}
```

Comment:
My understanding is this would be possible:

```yaml
        uses: google-github-actions/upload-cloud-storage@v2
        with:
          path: uploads/${{ steps.rename.outputs.PKG_FILE }}
          destination: ${{ steps.ftp_destination.outputs.FTP_DESTINATION }}
          headers: |-
            content-type: 'application/vnd.android.package-archive'
```

I'd be ok testing this first without though.